Rebus.AdoNet/AdoNetTimeoutStorage.cs

```diff
@@ -105,20 +148,33 @@ public DueTimeoutsResult GetDueTimeouts()
 				{
 					while (reader.Read())
 					{
-						var sqlTimeout = DueAdoNetTimeout.Create(MarkAsProcessed, timeoutsTableName, reader);
+						var sqlTimeout = DueAdoNetTimeout.Create(MarkAsProcessed, timeoutsTableName, reader, connection, transaction);
 
 						dueTimeouts.Add(sqlTimeout);
-					}
+					}
 				}
 			}
 
-			return new DueTimeoutsResult(dueTimeouts);
+			return new DueTimeoutsResult(dueTimeouts, () => CommitAndClose(connection, transaction));
+		}
+
+		void CommitAndClose(IDbConnection connection, IDbTransaction transaction)
+		{
+			try
+			{
+				transaction.Commit();
+				transaction.Dispose();
+				connection.Dispose();
+			}
+			catch (Exception ex)
+			{
+				log.Error("CloseAndCommit produced an exception: {0}", ex.Message);
+			}
 		}
 
 		void MarkAsProcessed(DueAdoNetTimeout dueTimeout)
 		{
-			using (var connection = factory.OpenConnection())
-			using (var command = connection.CreateCommand())
+			using (var command = dueTimeout.Connection.CreateCommand())
```

Comment:
This is missing lock, as to avoid two or more calls to MarkAsProcessed to use the same connection concurrently.